test/core/codec/leb128_test.cpp

```diff
@@ -0,0 +1,119 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <gtest/gtest.h>
+#include <boost/math/tools/precision.hpp>
+
+#include "codec/leb128/leb128.hpp"
+#include "primitives/boost_multiprecision.hpp"
+
+using namespace boost::multiprecision;
+using namespace boost::math::tools;
+
+/**
+ * @brief Check LEB128 encoding/decoding integrity
+ * @param data value to test
+ */
+template <typename T>
+bool LEB128IntegrityCheck(const T& data)
+{
+  std::vector<uint8_t> encoded = codec::leb128::encode(data);
+  T decoded = codec::leb128::decode<T>(encoded);
+  return data == decoded;
+}
+
+/**
+ * @given LEB128 encode/decode samples of the unsigned values
+ * @when Encoding value with LEB128 codec
+ * @then Produced values must be the same as the given samples
+ */
+TEST(LEB128_Algorithm, sample) {
+  uint32_t decoded{};
+  uint32_t source = 624485;
+  std::vector<uint8_t> sample{0xE5, 0x8E, 0x26};
+  std::vector<uint8_t> encoded = codec::leb128::encode(source);
+  ASSERT_EQ(sample, encoded);  // Check encoding
+  ASSERT_TRUE(true);
+  decoded = codec::leb128::decode<uint32_t>(encoded);
```

Comment:
Signature of the codec::leb128::decode<T> function was changed to
fc::outcome::result<T> decode<T>(const T &&data);
and move constructor will be called implicitly by compiler.
Also, uint32_t is a POD type, so it's trivially movable and C++ 11 and younger compiler first  of all will try to call move constructor instead of copy, so we don't need to force this call manually, yes?
Anyway, move semantics was forced in the appropriate calls, e.g.:
fc::outcome::result<T> &&decoded = decode<T>(std::move(encode(data)));